File: src/2965.find-missing-and-repeated-values.3227/solution.ts
function findMissingAndRepeatedValues (grid: number[][]): number[] {
    let axorb = 0;
    const n = grid.length;
    for (let i = 1; i <= n * n; i++) {
        axorb ^= i;
    }

    const set = new Set<number>();
    let a = 0;
    for (let i = 0; i < n; i++) {
        for (let j = 0; j < n; j++) {
            if (set.has(grid[i][j])) {
                a = grid[i][j];
            }
            axorb ^= grid[i][j];
            set.add(grid[i][j]);
        }
    }

    return [a, axorb ^ a, ];
}

File: src/2965.find-missing-and-repeated-values.3227/solution.java
class Solution {
    public int[] findMissingAndRepeatedValues(int[][] grid) {
        int N = grid.length;
        boolean[] exists = new boolean[N * N];
        int duplicate = 0;
        for (int[] row : grid) {
            for (int n : row) {
                if (exists[n - 1]) {
                    duplicate = n;
                }
                exists[n - 1] = true;
            }
        }
        int lost = 0;
        for (int i = 0; i < exists.length; i++) {
            if (!exists[i]) {
                lost = i + 1;
            }
        }
        return new int[] { duplicate, lost };
    }
}